assets/app/scripts/controllers/util/oauth.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:OAuthController
+ * @description
+ * # OAuthController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('OAuthController', function ($location, $q, RedirectLoginService, DataService, AuthService) {
+
+    var debug = true;
+
+    RedirectLoginService.finish()
+    .then(function(token, then) {
+      // Redirect to the root
+      if (!token) {
+        if (debug) { console.debug("OAuthController, no token or error, redirecting to /"); }
+        $location.url('/');
+        return;
+      }
+
+      // Try to fetch the user
+      var opts = {http: {auth: {token: token, triggerLogin: false}}};
+      if (debug) { console.debug("OAuthController, got token, fetching user", opts); }
+
+      DataService.get("users", "~", {}, opts)
+      .then(function(user) {
+        // Set the new user and token in the auth service
+        if (debug) { console.debug("OAuthController, got user", user); }
+        AuthService.setUser(user, token);
+
+        // Redirect to original destination (or default to '/')
+        var destination = then || '/';
+        if (debug) { console.debug("OAuthController, redirecting", destination); }
+        $location.url(destination);
```

Comment:
are we guaranteed that destination is a relative url? remember that $location doesn't work for absolute URLs